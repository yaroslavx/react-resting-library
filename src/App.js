// import { useEffect, useState } from 'react';

// function App() {
//   const [data, setData] = useState(null);
//   const [toggle, setToggle] = useState(false);
//   const [value, setValue] = useState('');

//   const handleToggle = () => {
//     setToggle((prev) => !prev);
//   };

//   useEffect(() => {
//     setTimeout(() => setData({ some: 'some' }), 100);
//   }, []);

//   return (
//     <div className='App'>
//       <h1 data-testid='input-value'>{value}</h1>
//       {toggle && <div data-testid='toggled-div'>Button is toggled </div>}
//       {data && <div style={{ color: 'red' }}>somedata</div>}
//       <h1>HI MY NAME IS</h1>
//       <button onClick={handleToggle} data-testid='toggle-button'>
//         SomeButton
//       </button>
//       <input
//         onChange={(e) => setValue(e.target.value)}
//         type='text'
//         placeholder='whats your name'
//       />
//     </div>
//   );
// }

// export default App;

import React from 'react';
import { Link, Route, Routes } from 'react-router-dom';
import Navbar from './components/Navbar';
import PageFirst from './pages/MainPage';
import PageNotFound from './pages/PageNotFound';
import PageSecond from './pages/SecondPage';
import UserPage from './pages/UserPage';
import AppRouter from './router/AppRouter';
import Users from './users/Users';

const App = () => {
  return (
    <>
      <Navbar />
      <AppRouter />
    </>
  );
};

export default App;
