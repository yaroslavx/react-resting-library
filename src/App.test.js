import { render, screen, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from './App';

describe('TEST APP', () => {
  test('render elems', () => {
    // render(<App />);
    // const hiMyNameIsEl = screen.getByText(/hi my name is/i);
    // const buttonEl = screen.getByRole('button');
    // const inputEl = screen.getByPlaceholderText(/whats your name/i);
    // expect(hiMyNameIsEl).toBeInTheDocument();
    // expect(buttonEl).toBeInTheDocument();
    // expect(inputEl).toMatchSnapshot();
  });
  // test('render elems', async () => {
  //   render(<App />);
  //   const hiMyNameIsEl = screen.queryByText(/hi my12345/i);
  //   expect(hiMyNameIsEl).toBeNull();
  //   const anotherHiMyNameIsEl = await screen.findByText(/somedata/i);
  //   expect(anotherHiMyNameIsEl).toBeInTheDocument();
  //   expect(anotherHiMyNameIsEl).toHaveStyle({ color: 'red' });
  // });
  // test('click event', async () => {
  //   render(<App />);
  //   const button = screen.getByTestId('toggle-button');
  //   expect(screen.queryByTestId('toggled-div')).toBeNull();
  //   fireEvent.click(button);
  //   expect(screen.queryByTestId('toggled-div')).toBeInTheDocument();
  // });
  // test('input event', async () => {
  //   render(<App />);
  //   const inputEl = screen.getByPlaceholderText(/whats your name/i);
  //   expect(screen.queryByTestId('input-value')).toContainHTML('');
  //   // fireEvent.input(inputEl, { target: { value: 'something' } });
  //   userEvent.type(inputEl, 'something');
  //   expect(screen.queryByTestId('input-value')).toContainHTML('something');
  // });
});
