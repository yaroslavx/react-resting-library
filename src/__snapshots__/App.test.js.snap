// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST APP render elems 1`] = `
<input
  placeholder="whats your name"
  type="text"
/>
`;
